src/creo2urdf/src/Sensorizer.cpp

```diff
@@ -0,0 +1,300 @@
+        }
+    }
+}
+
+std::vector<std::string> Sensorizer::buildSensorsXMLBlobs()
+{
+    std::vector<std::string> xml_blobs;
+
+    for (const auto& s : sensors)
+    {
+        xmlDocPtr doc = NULL;
+        xmlNodePtr root_node = NULL, node = NULL;
+        doc = xmlNewDoc(BAD_CAST "1.0");
+        root_node = xmlNewNode(NULL, BAD_CAST "gazebo");
+
+        xmlDocSetRootElement(doc, root_node);
+
+        xmlNewProp(root_node, BAD_CAST "reference", BAD_CAST s.linkName.c_str());
+
+        node = xmlNewChild(root_node, NULL, BAD_CAST "sensor", NULL);
+        xmlNewProp(node, BAD_CAST "name", BAD_CAST s.sensorName.c_str());
+
+        xmlNewProp(node, BAD_CAST "type", BAD_CAST gazebo_sensor_type_map.at(s.type).c_str());
+
+        xmlNewChild(node, NULL, BAD_CAST "always_on", BAD_CAST "1");
+        xmlNewChild(node, NULL, BAD_CAST "update_rate", BAD_CAST to_string(s.updateRate).c_str());
+
+        iDynTree::Transform trf = s.transform;
+
+        string pose = trf.getPosition().toString() + " " + trf.getRotation().asRPY().toString();
+
+        xmlNewChild(node, NULL, BAD_CAST "pose", BAD_CAST pose.c_str());
+
+        for (auto blob : s.xmlBlobs)
+        {
+            blob.erase(std::remove_if(blob.begin(), blob.end(),
+                [](unsigned char c) {
+                    return !std::isprint(c);
+                }),
+                blob.end());
+
+            xmlNodePtr node_xmlblob = nullptr;
+
+            xmlParseInNodeContext(node, blob.c_str(), blob.size(), 0, &node_xmlblob);
+
+            if (node_xmlblob)
+                xmlAddChild(node, node_xmlblob);
+        }
+
+        xmlOutputBufferPtr doc_buffer = xmlAllocOutputBuffer(NULL);
+        xmlNodeDumpOutput(doc_buffer, doc, root_node, 0, 1, NULL);
+        xml_blobs.push_back(string((char*)xmlBufContent(doc_buffer->buffer)));
+
+        xmlOutputBufferClose(doc_buffer);
+
+        xmlFreeDoc(doc);
+        xmlCleanupParser();
+
+        doc = xmlNewDoc(BAD_CAST "1.0");
+        root_node = xmlNewNode(NULL, BAD_CAST "sensor");
+        xmlDocSetRootElement(doc, root_node);
+
+        xmlNewProp(root_node, BAD_CAST "name", BAD_CAST s.sensorName.c_str());
+        xmlNewProp(root_node, BAD_CAST "type", BAD_CAST sensor_type_map.at(s.type).c_str());
+        node = xmlNewChild(root_node, NULL, BAD_CAST "parent", NULL);
+        xmlNewProp(node, BAD_CAST "link", BAD_CAST s.linkName.c_str());
+        node = xmlNewChild(root_node, NULL, BAD_CAST "origin", NULL);
+        xmlNewProp(node, BAD_CAST "rpy", BAD_CAST trf.getRotation().asRPY().toString().c_str());
+        xmlNewProp(node, BAD_CAST "xyz", BAD_CAST trf.getPosition().toString().c_str());
+
+        doc_buffer = xmlAllocOutputBuffer(NULL);
+        xmlNodeDumpOutput(doc_buffer, doc, root_node, 0, 1, NULL);
+        xml_blobs.push_back(string((char*)xmlBufContent(doc_buffer->buffer)));
+
+        xmlOutputBufferClose(doc_buffer);
+
+        xmlFreeDoc(doc);
+        xmlCleanupParser();
```

Comment:
Please remove, this function needs to be called only before exit (see https://gnome.pages.gitlab.gnome.org/libxml2/devhelp/libxml2-parser.html#xmlCleanupParser and https://gnome.pages.gitlab.gnome.org/libxml2/devhelp/libxml2-parser.html#xmlCleanupParser).